### ISSUE TITLE
Issue with Heading wrapped in link

### ISSUE THREAD

**Issue by victory-sokolov** [crosstype/node-html-markdown#46]
Hey! First of all would like to thank you for such a great library!

Noticed a few issues when generating Markdown.

I have the following HTML:

```html
<a href="/post/some-post">
  <h2>Subheading</h2>
 </a>
```

It produces the following Markdown

```
[Subheading](/post/some-post)
```

Expected output would be:

```
## [Subheading](/post/some-post)
```

Another issue I've encountered is that multiline code is not wrapped to the backticks, but i guess i can open another issue for this.

**Comment by nonara** [comment]
Hi Viktor. Thank you for the report & kind words.

You're right that this isn't ideal. I will give some thought on how to approach this.

Can you give some more detail on the multiline code issue (an example, preferably). Code fencing should be working properly.

**Comment by victory-sokolov** [comment]
Yes, sure here is an example of the HTML and when converting to markdown it converts to plain text without wrapping code into backticks

<pre class="prism-code language-typescript" style="color: rgb(248, 248, 242); background-color: rgb(40, 42, 54);"><div class="sc-36549842-4 bmkFVs"></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">1</span><span class="sc-36549842-2 fODbQc"><span class="token keyword" style="color: rgb(189, 147, 249); font-style: italic;">class</span><span class="token plain"> </span><span class="token class-name">Invoice</span><span class="token plain"> </span><span class="token punctuation" style="color: rgb(248, 248, 242);">{</span><span class="token plain"></span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">2</span><span class="sc-36549842-2 fODbQc"><span class="token plain">    </span><span class="token function" style="color: rgb(80, 250, 123);">constructor</span><span class="token punctuation" style="color: rgb(248, 248, 242);">(</span><span class="token keyword" style="color: rgb(189, 147, 249); font-style: italic;">public</span><span class="token plain"> persistanceStorage</span><span class="token operator">:</span><span class="token plain"> PersistanceStorage</span><span class="token punctuation" style="color: rgb(248, 248, 242);">)</span><span class="token plain"> </span><span class="token punctuation" style="color: rgb(248, 248, 242);">{</span><span class="token punctuation" style="color: rgb(248, 248, 242);">}</span><span class="token plain"></span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">3</span><span class="sc-36549842-2 fODbQc"><span class="token plain" style="display: inline-block;">
</span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">4</span><span class="sc-36549842-2 fODbQc"><span class="token plain">    </span><span class="token keyword" style="color: rgb(189, 147, 249); font-style: italic;">public</span><span class="token plain"> </span><span class="token keyword" style="color: rgb(189, 147, 249); font-style: italic;">async</span><span class="token plain"> </span><span class="token function" style="color: rgb(80, 250, 123);">persistInvoice</span><span class="token punctuation" style="color: rgb(248, 248, 242);">(</span><span class="token plain">data</span><span class="token punctuation" style="color: rgb(248, 248, 242);">)</span><span class="token plain"> </span><span class="token punctuation" style="color: rgb(248, 248, 242);">{</span><span class="token plain"></span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">5</span><span class="sc-36549842-2 fODbQc"><span class="token plain">  </span><span class="token keyword" style="color: rgb(189, 147, 249); font-style: italic;">this</span><span class="token punctuation" style="color: rgb(248, 248, 242);">.</span><span class="token plain">persistanceStorage</span><span class="token punctuation" style="color: rgb(248, 248, 242);">.</span><span class="token function" style="color: rgb(80, 250, 123);">saveData</span><span class="token punctuation" style="color: rgb(248, 248, 242);">(</span><span class="token plain">data</span><span class="token punctuation" style="color: rgb(248, 248, 242);">)</span><span class="token punctuation" style="color: rgb(248, 248, 242);">;</span><span class="token plain"></span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">6</span><span class="sc-36549842-2 fODbQc"><span class="token plain">    </span><span class="token punctuation" style="color: rgb(248, 248, 242);">}</span><span class="token plain"></span></span></div><div class="sc-36549842-0 kfTyDW token-line" style="color: rgb(248, 248, 242);"><span class="sc-36549842-1 eePLWY">7</span><span class="sc-36549842-2 fODbQc"><span class="token plain"></span><span class="token punctuation" style="color: rgb(248, 248, 242);">}</span></span></div></pre>

**Comment by nonara** [comment]
Ah, ok. Thanks for the example! Yes, we do handle `pre` by itself differently than `pre` with `code`, to best handle all situations.

Try wrapping with `code` (inside `pre`), and it should work the way you'd like!

**Comment by victory-sokolov** [comment]
Thank you, wrapping inside the `code`, worked well!
